src/driver-hidpp20.c

```diff
@@ -1000,6 +1000,14 @@ hidpp20drv_update_resolution_dpi(struct ratbag_resolution *resolution,
 	int i;
 	int dpi = dpi_x; /* dpi_x == dpi_y if we don't have the individual resolution cap */
 
+	if (ratbag_resolution_has_capability(resolution, RATBAG_RESOLUTION_CAP_DISABLE)) {
+		if (resolution->is_disabled)
+			dpi = dpi_x = dpi_y = 0;
+	} else {
+		if (resolution->is_disabled)
+			return -ENOTSUP;
+	}
```

Comment:
nitpick: make `if (resolution->is_disabled)` the outer if

Reply:
But why are we returning if the resolution doesn't have the disable capability and `resolution->is_disabled` is true? If the resolution doesn't have that capability `resolution->is_disabled` should be ignored.

Reply:
> nitpick: make if (resolution->is_disabled) the outer if

Ah yep that makes sense.

> If the resolution doesn't have that capability resolution->is_disabled should be ignored.

But it shouldn't be set as well. If someone sets the resolution to disabled even if it's not supported (I think that theoretically possible with DBus calls, not with ratbagctl or Piper though), we should throw a not-supported-error.
Anyway, if we swap the if-statements it's basically free (just an else).